crates/sui-core/src/authority.rs

```diff
@@ -2887,6 +2880,11 @@ impl AuthorityState {
         }
     }
 
+    pub async fn execution_lock_for_signing(&self) -> ExecutionLockReadGuard {
+        // No need to check epoch here - current epoch is always correct for signing
```

Comment:
Could you comment explain why signing also requires execution lock?

Reply:
Did you read the PR description? It's to prevent reconfiguration from happening while we are signing a tx.

Reply:
Yes, but please also doc comment it.

Reply:
oh of course, I have no idea why i thought you meant "comment in the PR"